File: ReduxMVI/src/main/java/com/krykun/reduxmvi/log/jlogger/JLoggerFactory.java
package com.krykun.reduxmvi.log.jlogger;

public class JLoggerFactory {

    private JLoggerFactory() {
        throw new IllegalStateException("JLoggerFactory class");
    }

    public static JLogger getLogger(Class<?> clazz) {
        return new JLogger(clazz.getSimpleName());
    }
}

File: ReduxMVI/src/main/java/com/krykun/reduxmvi/log/jlogger/JLoggerFactory.kt
package com.krykun.reduxmvi.log.jlogger

class JLoggerFactory private constructor() {
    init {
        throw IllegalStateException("JLoggerFactory class")
    }

    companion object {
        fun getLogger(clazz: Class<*>): JLogger {
            return JLogger(clazz.simpleName)
        }
    }
}